NitroxClient/Communication/MultiplayerSession/MultiplayerSessionManager.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.Communication.MultiplayerSession.ConnectionState;
+using NitroxModel.Logger;
+using NitroxModel.MultiplayerSession;
+using NitroxModel.Packets;
+
+namespace NitroxClient.Communication.MultiplayerSession
+{
+    public class MultiplayerSessionManager : IMultiplayerSession, IMultiplayerSessionConnectionContext
+    {
+        private readonly HashSet<Type> suppressedPacketsTypes = new HashSet<Type>();
+        private readonly IClient client;
+        private string ipAddress;
+        private MultiplayerSessionPolicy sessionPolicy;
+        private PlayerSettings playerSettings;
+        private AuthenticationContext authenticationContext;
+
+        IClient IMultiplayerSessionState.Client => client;
+        string IMultiplayerSessionState.IpAddress => ipAddress;
+        MultiplayerSessionPolicy IMultiplayerSessionState.SessionPolicy => sessionPolicy;
+        PlayerSettings IMultiplayerSessionState.PlayerSettings => playerSettings;
+        AuthenticationContext IMultiplayerSessionState.AuthenticationContext => authenticationContext;
+        public MultiplayerSessionReservation Reservation { get; private set; }
+
+        public MultiplayerSessionManager(IClient client)
+        {
+            Log.Info("Initializing MultiplayerSessionManager...");
+            this.client = client;
+            CurrentState = new Disconnected();
+        }
+
+        //Testing entry point
+        internal MultiplayerSessionManager(IClient client, IMultiplayerSessionConnectionState initialState)
+        {
+            this.client = client;
+            CurrentState = initialState;
+        } 
+
+        public event MultiplayerSessionConnectionStateChangedEventHandler ConnectionStateChanged;
+        public IMultiplayerSessionConnectionState CurrentState { get; private set; }
+
+        //Eww...
+        public string PlayerId { get; private set; }
```

Comment:
This actually needs to be deprecated. I split player ID and player Name. The player ID is now provided on the session reservation object.